src/main/java/org/generationcp/middleware/dao/AttributeDAO.java

```diff
@@ -88,4 +91,91 @@ public Attribute getAttribute(final Integer gid, final String attributeName) {
 		}
 		return attribute;
 	}
+
+	public List<AttributeDTO> getAttributesByGidAndAttributeIds(
+		final String gid, final List<String> attributeIds, final Integer pageSize, final Integer pageNumber) {
+
+		List<AttributeDTO> attributes;
+		try {
+			String sql = "SELECT "
```

Comment:
@vmaletta  You can make this a constant and do a select count(1) from the declared constant and we will have only one query to maintain